pkgs/tools/system/plan9port/default.nix

```diff
@@ -19,26 +19,67 @@ stdenv.mkDerivation rec {
   postPatch = ''
     substituteInPlace bin/9c \
       --replace 'which uniq' '${which}/bin/which uniq'
+    ed -sE INSTALL <<EOF
+    # get /bin:/usr/bin out of PATH
+    /^PATH=[^ ]*/s,,PATH=\$PATH:\$PLAN9/bin,
+    # no xcbuild nonsense
+    /^if.* = Darwin/+;/^fi/-c
+    ${"\t"}export NPROC=$NIX_BUILD_CORES
+    .
+    # remove absolute include paths from fontsrv test
+    /cc -o a.out -c -I.*freetype2/;/x11.c/j
+    s/(-Iinclude).*-I[^ ]*/\1/
+    wq
+    EOF
   '';
 
-  buildInputs = [ perl ] ++ (if !stdenv.isDarwin then [
-    fontconfig
-    freetype # fontsrv wants ft2build.h
-    libX11
-    libXext
-    libXt
-    xorgproto
+  buildInputs = [ perl which ed ] ++ (if !stdenv.isDarwin then [
```

Comment:
Maybe `which` and `ed` are not used in the runtime, so you can put them in `nativeBuildInputs`.

Reply:
As described above, `which` definitely belongs in `buildInputs`, but yes, `ed` should go in `nativeBuildInputs` now that you mention it.